app/models/library_entry.rb

```diff
@@ -209,6 +209,21 @@ def kind
       user.update_feed_completed!
       user.update_profile_completed!
     end
+
+    if progress_changed?
+      case kind
+      when :anime
+        guess = [(progress + 1), anime.default_progress_limit].min
+        anime.update_episode_count_guess(guess) if
```

Comment:
The `guess` line could probably be unified for all media types as `guess = [(progress + 1), media.default_progress_limit].min`, but it'd be nice to get `media.update_unit_count_guess(guess)` too.

I generally try to make it a rule of keeping as many method names common between media types as possible, so that we don't need conditionals to control behavior (it's just inherent as part of the different model type)